src/NServiceBus.Persistence.NonDurable/SubscriptionStorage/NonDurableSubscriptionStorage.cs

```diff
@@ -30,17 +31,14 @@ public Task Unsubscribe(Subscriber subscriber, MessageType messageType, ContextB
 
         public Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes, ContextBag context, CancellationToken cancellationToken = default)
         {
-            var result = new HashSet<Subscriber>();
-            foreach (var m in messageTypes)
-            {
-                if (storage.TryGetValue(m, out var list))
-                {
-                    result.UnionWith(list.Values);
-                }
-            }
-            return Task.FromResult((IEnumerable<Subscriber>)result);
+            var subscribers = messageTypes
```

Comment:
While this code could have returned duplicate subscribers, there's code in NServiceBus's UnicastPublishRouter class in SelectDestinationsForEachEndpoint() that dedupes destination addresses, making this more of a refactoring, and not a bug that needs to be patched.